cpu/esp8266/include/cpu_conf.h

```diff
@@ -69,6 +69,14 @@ extern "C" {
 #ifndef TCPIP_THREAD_STACKSIZE
 #define TCPIP_THREAD_STACKSIZE        (3072)
 #endif
+
+#ifndef GCOAP_STACK_SIZE
+#ifdef MODULE_GNRC
+#define GCOAP_STACK_SIZE              (1536 + DEBUG_EXTRA_STACKSIZE)
+#elif MODULE_LWIP
+#define GCOAP_STACK_SIZE              (2048 + DEBUG_EXTRA_STACKSIZE)
```

Comment:
I know this is floating around for a while now, but any idea why this is? Maybe this is explained by data on the stack that rather should not be there?

Reply:
Excellent question. Beyond the measurements in #13606 I also checked if the stack use increase for a CoAP request was on the request send side or the response receive side. At least for GNRC, the increase is on receive side.

I have not dug much into the ESP platform internals to account for the differences. One thing I did notice for the lwIP stack is that the `netconn` attribute used in the lwip `sock_udp` struct includes several attributes.

@gschorcht any information or even intuition here for the increased stack use in the gcoap thread for the GNRC or lwIP stacks?